### PR TITLE
[Term Entry] Lua tables concat()

### PR DESCRIPTION
**Pull request by phil-haywood**
### Description

-Added the tables concat() term for Lua
-Closes issue #3115 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/lua/concepts/tables/terms/concat/concat.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  - 'Tables'
+  - 'Functions'
+CatalogContent:
+  - 'learn-lua'
+  - 'paths/computer-science'
+---
+
+# `concat()` Function
+
+## Introduction
```
Comment (caupolicandiaz) [inline content/lua/concepts/tables/terms/concat/concat.md:17]: entries do not begin with headers, delete both

File: content/lua/concepts/tables/terms/concat/concat.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+The `concat()` function is used to join the values of a table into a single string, separated by a specified separator. This function is particularly useful when you need to convert a table into a formatted string representation. It allows you to customize the separator between values and optionally specify a range of indices to concatenate.
```
Comment (caupolicandiaz) [inline content/lua/concepts/tables/terms/concat/concat.md:18]:
```suggestion
The **`concat()`** function is used to join the values of a table into a single string, separated by a specified separator. This function is particularly useful when you need to convert a table into a formatted string representation. It allows you to customize the separator between values and optionally specify a range of indices to concatenate. 

```

File: content/lua/concepts/tables/terms/concat/concat.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  - 'learn-lua'
+  - 'paths/computer-science'
+---
+
+# `concat()` Function
+
+## Introduction
+The `concat()` function is used to join the values of a table into a single string, separated by a specified separator. This function is particularly useful when you need to convert a table into a formatted string representation. It allows you to customize the separator between values and optionally specify a range of indices to concatenate. 
+
+## Syntax
```
Comment (caupolicandiaz) [inline content/lua/concepts/tables/terms/concat/concat.md:20]: add an empty line after the header

File: content/lua/concepts/tables/terms/concat/concat.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+```lua
```
Comment (caupolicandiaz) [inline content/lua/concepts/tables/terms/concat/concat.md:21]:
```suggestion
```pseudo

```

File: content/lua/concepts/tables/terms/concat/concat.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+print(result)
+```
+
+The code will return the following output:
+
+```shell
+banana 🍌, cherry 🍒
+```
+
+In this example, we have a table `fruits` containing four elements. We use `table.concat()` to join the elements from index 2 to 3 (inclusive) with a comma and space as the separator. The resulting string is then printed, which contains "banana 🍌" and "cherry 🍒" separated by a comma and space.
```
Comment (caupolicandiaz) [inline content/lua/concepts/tables/terms/concat/concat.md:41]:
```suggestion
In this example, the table `fruits` contains four elements. `table.concat()` is used to join the elements from index 2 to 3 (inclusive) with a comma and space as the separator. The resulting string is then printed.

```

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@phil-haywood I've added my initial comments. Please update the entry, and tag me when you've finished making changes. Thanks

**Comment by phil-haywood** [comment]
@caupolicandiaz I think I have everything updated. Let me know if there is anything else. Thanks

File: content/lua/concepts/tables/terms/concat/concat.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  - 'Game Development'
+Tags:
+  - 'Tables'
+  - 'Functions'
+CatalogContent:
+  - 'learn-lua'
+  - 'paths/computer-science'
+---
+
+The **`concat()`** function is used to join the values of a table into a single string, separated by a specified separator. This function is particularly useful when you need to convert a table into a formatted string representation. It allows you to customize the separator between values and optionally specify a range of indices to concatenate.
```
Comment (caupolicandiaz) [inline content/lua/concepts/tables/terms/concat/concat.md:15]:
```suggestion
The **`concat()`** function is used to join the values of a table into a single string, separated by a specified separator. This function is particularly valuable when a table needs to be converted into a formatted string representation. The method allows for the use of a custom separator between values and the option to specify a range of indices to concatenate.
```

Comment (caupolicandiaz) [reply content/lua/concepts/tables/terms/concat/concat.md:15]: docs avoids the use of first and second person pronouns (I, you, we)

File: content/lua/concepts/tables/terms/concat/concat.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+## Syntax
+
+```pseudo
+concat(table, separator, [i], [j])
+```
```
Comment (caupolicandiaz) [inline content/lua/concepts/tables/terms/concat/concat.md:21]: there should be descriptions for the parameters below the syntax block

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@phil-haywood I added a few additional comments that I had overlooked in my initial review.

**Comment by phil-haywood** [comment]
@caupolicandiaz I believe I have it now. 😊

**Review by caupolicandiaz — APPROVED** [review]
@phil-haywood I made a couple of direct changes to the parameter section. This is ready for a second review.

**Comment by phil-haywood** [comment]
@caupolicandiaz thank you

File: content/lua/concepts/tables/terms/concat/concat.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+Consider the following table:
+
+```lua
+local fruits = {"apple 🍎", "banana 🍌", "cherry 🍒", "watermelon 🍉"}
+local result = table.concat(fruits, ", ", 2, 3)
+print(result)
+```
+
+The code will return the following output:
```
Comment (KTom101) [inline content/lua/concepts/tables/terms/concat/concat.md:42]:
```suggestion
The output is the following:
```

**Review by KTom101 — CHANGES_REQUESTED** [review]
@phil-haywood I added a few changes to this. You will also have to remove the `yarn.lock` file.

**Comment by caupolicandiaz** [comment]
@phil-haywood @KTom101 For the sake of moving this along I've gone ahead and made the changes to the entry

**Comment by phil-haywood** [comment]
@caupolicandiaz  @KTom101 OK, thanks.

**Review by KTom101 — APPROVED** [review]

**Pull request merged**